### PR TITLE
Test backfill

### PR DESCRIPTION
**Pull request by AgeManning**
Adds a feature flag in able to shorten the backfill distance for developers to help debug and diagnose issues with backfill syncing.

### CODE REVIEW

File: beacon_node/beacon_chain/src/builder.rs [hunk 1]
```diff
@@ -832,10 +832,14 @@ where
         let genesis_backfill_slot = if self.chain_config.genesis_backfill {
             Slot::new(0)
         } else {
+            #[cfg(not(test_backfill))]
```
Comment (jxs) [inline beacon_node/beacon_chain/src/builder.rs]: should be `#[cfg(not(feature = "test_backfill")]` right?
btw, nit: you can do:
```suggestion
            let backfill_epoch_range = if cfg!(feature = "test_backfill") {
                3
            } else {
                (self.spec.min_validator_withdrawability_delay + self.spec.churn_limit_quotient)
                    .as_u64()
                    / 2
            };

```

**Comment by AgeManning** [comment]
oh nice! Thanks!

File: beacon_node/beacon_chain/src/builder.rs [hunk 2]
```diff
@@ -832,10 +832,21 @@ where
...
+                    / 2
+            };
+
             let backfill_epoch_range = (self.spec.min_validator_withdrawability_delay
                 + self.spec.churn_limit_quotient)
                 .as_u64()
                 / 2;
+            #[cfg(test_backfill)]
+            let backfill_epoch_range = 3;
+
```
Comment (jxs) [inline beacon_node/beacon_chain/src/builder.rs]:
```suggestion
```

this is no longer required right?

**Comment by AgeManning** [comment]
@mergifyio queue

**Review by jxs — APPROVED** [review]

**Comment by AgeManning** [comment]
@mergifyio requeue

**Pull request merged**